src/cpp/common/alts_context.cc

```diff
@@ -0,0 +1,44 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "alts_context.h"
+
+#include <grpc/grpc_security.h>
+
+#include "src/core/tsi/alts/handshaker/alts_tsi_handshaker.h"
+#include "src/cpp/common/secure_auth_context.h"
+
+namespace grpc {
+std::unique_ptr<gcp::AltsContext> GetAltsContextFromAuthContext(
+    const AuthContext& auth_context) {
+  std::unique_ptr<gcp::AltsContext> uniq_ctx(new gcp::AltsContext());
```

Comment:
It is better to return nullptr in the case of no AltsContext or error. So move this line after line 35.
You can return nullptr in line 34.